src/common/components/OnlyFor.jsx

```diff
@@ -1,40 +1,44 @@
 import { memo } from 'react';
 import PropTypes from 'prop-types';
-import { Box, Button, useColorModeValue } from '@chakra-ui/react';
+import { Box, Button } from '@chakra-ui/react';
 import { useRouter } from 'next/router';
 import useTranslation from 'next-translate/useTranslation';
 import Icon from './Icon';
+import useStyle from '../hooks/useStyle';
 
 const OnlyFor = ({
-  cohortSession, academy, capabilities, children, onlyMember, withBanner,
+  cohortSession, academy, capabilities, children, onlyMember, withBanner, profile,
 }) => {
   const { t } = useTranslation('common');
 
   const academyNumber = Math.floor(academy);
   const router = useRouter();
-  const userCapabilities = cohortSession.user_capabilities || [];
-  const commonUser = ['STUDENT', 'REVIEWER'];
-  const cohortRole = cohortSession.cohort_role?.toUpperCase() || 'NONE';
-  const featuredBackground = useColorModeValue('featuredLight', 'featuredDark');
-  const backgroundColor = useColorModeValue('white', 'darkTheme');
+  const userCapabilities = profile.permissionsSlug || [];
+  const commonUser = ['TEACHER', 'ASSISTANT', 'STUDENT', 'REVIEWER'];
+  const profileRole = profile?.roles?.length > 0 && profile?.roles[0]?.role?.toUpperCase();
+  const cohortRole = cohortSession?.cohort_role?.toUpperCase() || profileRole || 'NONE';
+  const { featuredColor, backgroundColor } = useStyle();
 
   const isCapableAcademy = cohortSession && cohortSession.academy?.id === academyNumber;
+  const isMember = commonUser.includes(cohortRole);
+  const capabilitiesNotExists = capabilities.length > 0 && capabilities.includes('');
```

Comment:
the name of the variable should be `capablitiesExists` because it returns true if they exist

Reply:
`capabilities` have an array by default, this also has the purpose of conditioning the existence of an array with empty string